store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -117,50 +118,63 @@ public Value getValueFromElement(final Element element) {
-                    if (currentPropLength > 0) {
-                        try {
-                            properties.put(propertyName, serialiser.deserialise(Arrays.copyOfRange(bytes, lastDelimiter, lastDelimiter += currentPropLength)));
-                        } catch (final SerialisationException e) {
-                            throw new AccumuloElementConversionException("Failed to deserialise property " + propertyName, e);
-                        }
-                    } else {
-                        try {
-                            properties.put(propertyName, serialiser.deserialiseEmpty());
-                        } catch (final SerialisationException e) {
-                            throw new AccumuloElementConversionException("Failed to deserialise property " + propertyName, e);
+        if (isNotEmpty(value)) {
+            final byte[] bytes = value.get();
+            int delimiterPosition = 0;
+            final int arrayLength = bytes.length;
+            final SchemaElementDefinition elementDefinition = getSchemaElementDefinition(group);
+            final Iterator<String> propertyNames = elementDefinition.getProperties().iterator();
+            while (propertyNames.hasNext() && delimiterPosition < arrayLength) {
+                final String propertyName = propertyNames.next();
+                try {
+                    if (isStoredInValue(propertyName, elementDefinition)) {
+                        final TypeDefinition typeDefinition = elementDefinition.getPropertyTypeDef(propertyName);
+                        if (typeDefinition != null) {
+                            ToBytesSerialiser serialiser = (ToBytesSerialiser) typeDefinition.getSerialiser();
+                            if (serialiser != null) {
+                                byte delimiterByte = bytes[delimiterPosition];
+                                final int numBytesForLength = CompactRawSerialisationUtils.decodeVIntSize(delimiterByte);
+                                final long currentPropLength = getCurrentPropLength(bytes, delimiterPosition, numBytesForLength);
+                                Object deserialisedObject = getDeserialisedObject(serialiser, bytes, delimiterPosition + numBytesForLength, currentPropLength);
+                                properties.put(propertyName, deserialisedObject);
+                                delimiterPosition += numBytesForLength + currentPropLength;
+                            }
                         }
                     }
+                } catch (final SerialisationException e) {
+                    throw new AccumuloElementConversionException("Failed to deserialise property " + propertyName, e);
                 }
             }
         }
-
         return properties;
     }
 
+    private Object getDeserialisedObject(final ToBytesSerialiser serialiser, final byte[] bytes, final int lastDelimiter, final long currentPropLength) throws SerialisationException {
```

Comment:
Minor issue but if possible please could you move these 2 private methods to the bottom of the class.